sympy/integrals/manualintegrate.py

```diff
@@ -898,6 +899,31 @@ def root_mul_rule(integral):
         return URule(u, u_func, None, next_step, integrand, symbol)
 
 
+def mul_pow_derivative_rule(integral: tuple[Expr, Symbol]):
+    """
+    Apply "Integral(x**n*Derivative(f(x), (x, n + 1)), x) equals to x**n*Derivative(f(x), (x, n))
+    - n*Integral(x**(n - 1)*Derivative(f(x), (x, n)), x)" n times
```

Comment:
Is it possible to use Cauchy's formula rather than repeated integration here?

https://en.wikipedia.org/wiki/Cauchy_formula_for_repeated_integration

Reply:
Here we need all intermediate $f^{(n-i)}(x)$, not only for constructing final result $\Sigma_{i=0}^n (-1)^{i}\frac{n!}{(n-i)!}x^{n-i} f^{(n-i)}(x)$, but also for showing manual steps, so the formula doesn't seem applicable here.

Reply:
I wondered if it might be more efficient to compute the nth integral once and then differentiate it to get the intermediate results. Maybe the integral implied by the formula would be harder than the original integral though...

Reply:
How well does this perform if `n` is large?

Reply:
> Maybe the integral implied by the formula would be harder than the original integral though...

That's the case. This PR assumes `x^n * f(x)` is hard to integrate, and if `f(x)` can be integrated `n+1` times, then `x^n * f(x)` can be computed. The Cauchy formula assumes a definite integration `(x-t)^n * f(t)` is easy to compute, so it can be used to integrate `f(x)` `n+1` times. That's just the inverse problem.

> How well does this perform if `n` is large?

If `n` is large, $f^{(n+1)}(x)$ tends to be complicated, so it's likely not be integrated by existing algorithms.
```py
>>> from sympy import *
>>> x=Symbol('x')
>>> [integrate(tan(x).diff(x, n), x, manual=True) for n in range(4)]
[-log(cos(x)), tan(x), tan(x)**2, -6*x + 8*tan(x) + 6*Integral(tan(x)**4, x)]
```

Reply:
I guess it's not worth using the formula.

As an aside I just tried this and it's really slow (on master, due to heurisch):
```python
In [6]: %time integrate(x**20*sin(x))
CPU times: user 1min 16s, sys: 116 ms, total: 1min 16s
Wall time: 1min 16s
Out[6]: 
   20              19               18                17                  16                   15  
- x  ⋅cos(x) + 20⋅x  ⋅sin(x) + 380⋅x  ⋅cos(x) - 6840⋅x  ⋅sin(x) - 116280⋅x  ⋅cos(x) + 1860480⋅x  ⋅s

                  14                     13                      12                       11       
in(x) + 27907200⋅x  ⋅cos(x) - 390700800⋅x  ⋅sin(x) - 5079110400⋅x  ⋅cos(x) + 60949324800⋅x  ⋅sin(x)

                 10                         9                          8                           
 + 670442572800⋅x  ⋅cos(x) - 6704425728000⋅x ⋅sin(x) - 60339831552000⋅x ⋅cos(x) + 482718652416000⋅x

7                            6                             5                              4        
 ⋅sin(x) + 3379030566912000⋅x ⋅cos(x) - 20274183401472000⋅x ⋅sin(x) - 101370917007360000⋅x ⋅cos(x) 

                      3                               2                                            
+ 405483668029440000⋅x ⋅sin(x) + 1216451004088320000⋅x ⋅cos(x) - 2432902008176640000⋅x⋅sin(x) - 243

                       
2902008176640000⋅cos(x)

```